package.json

```diff
@@ -10,7 +10,7 @@
     "prod": "cross-env NODE_ENV=production pm2-runtime start ./build/server.js",
     "start": "cross-env pm2-runtime start ./build/server.js --watch",
     "setup": "tsx setup.ts",
-    "test": "vitest run --pool forks --poolOptions.forks.singleFork --coverage",
+    "test": "vitest run --pool=threads --no-file-parallelism --coverage",
```

Comment:
parallelism is fine but may also lead to race conditions and make tests flaky. I suggest to make sure to keep test files separately that depends on the other tests to run in same thread

Reply:
If you can create a new ticket for above

Reply:
Thanks for the feedback, @varshith257!  🙌
I've created ticket #2491 to track any further issues with parallelism and race conditions for ongoing monitoring and improvements.